tests/xbm/github_bug_170.c

```diff
@@ -0,0 +1,41 @@
+/*
+	Test writing of XBM images with a width that is not a multiple of 8
+
+	We create an image with a width of 11 pixels, and draw a circle on it.
+	To test that the padding is correctly applied, we write the image to disk
+	and assert that the number of bytes written matches our expectation.
+
+	See also <https://github.com/libgd/libgd/issues/170>.
+*/
+
+#include "gd.h"
+#include "gdtest.h"
+
+int main()
+{
+	gdImagePtr im;
+	int black;
+	FILE *outFile;
+	gdIOCtx *out;
+	off_t length;
+
+	/* create the test image */
+	im = gdImageCreate(11, 11);
+	gdImageColorAllocate(im, 255, 255, 255);
+	black = gdImageColorAllocate(im, 0, 0, 0);
+	gdImageArc(im, 5, 5, 10, 10, 0, 360, black);
+
+	/* write the file to disk, note the file length and delete the file */
+	outFile = gdTestTempFp();
+	out = gdNewFileCtx(outFile);
+	gdTestAssert(out != NULL);
+	gdImageXbmCtx(im, "github_bug_170.xbm", 1, out);
+	out->gd_free(out);
+	length = ftello(outFile);
+	fclose(outFile);
+
+	gdImageDestroy(im);
+
+	gdTestAssertMsg(length == 250, "expected to write 250 bytes; %d bytes written", length);
```

Comment:
According to a [StackOverflow answer](http://stackoverflow.com/questions/586928/how-should-i-print-types-like-off-t-and-size-t#answer-22537979) `%jd` is not supported by VS 2013. Wouldn't that be an issue?

Also I'm not sure about `intmax_t`. This appears to be defined in `inttypes.h`, but I'm not sure if that's portable (especially considering MSVC).

Still have not been able to build libgd on Windows.